src/libstd/panicking.rs

```diff
@@ -29,12 +29,45 @@ use intrinsics;
 use mem;
 use ptr;
 use raw;
-use sys::stdio::{Stderr, stderr_prints_nothing};
+use sys::stdio::Stderr;
 use sys_common::rwlock::RWLock;
 use sys_common::thread_info;
 use sys_common::util;
 use thread;
 
+pub(crate) trait PanicWrite {
+    /// Is there any way to output panic messages on this platform? If this
+    /// returns `false`, callers might avoid calling `write`. However, `write`
+    /// may still be called and implementors must be prepared for this.
+    fn is_enabled() -> bool;
+
+    /// This function should construct a `Write`r and call the closure passing
+    /// it in.
+    fn write<T, F: FnOnce(&mut dyn Write) -> T>(write: F);
+}
+
+pub use sys::stdio::PanicOutput;
```

Comment:
Instead of this and a trait, could this just be a global function that's something along the lines of `fn panic_output() -> Option<impl Write>`? (each platform doesn't even need to return the same type)

Reply:
I think I had something like that but you requested I change that in order to keep [this line](https://github.com/rust-lang/rust/pull/56066/files#diff-dec9d6f114f706b628f5fcc9222f739dR502).

Reply:
Right yeah, but you previously had something like `fn() -> Option<fn() -> impl Write>`, right? Could this simply be roughly `fn() -> Option<impl Write>`?

Reply:
Done